Add test for IPv6 functionality

mcipperly opened the pull request:
I wrote a test for PR #138 which adds IPv6 support ( #137 )

mcipperly commented:
Apologies for the spam, my git is rusty and I haven't built on another PR previously!

Travis doesn't seem to support IPv6 (more info [here](https://docs.travis-ci.com/user/reference/overview/https://docs.travis-ci.com/user/reference/overview/#virtualisation-environment-vs-operating-system) and [here](https://github.com/njh/travis-ipv6-test) ) so I did add a condition to skip in that case. Unfortunately this causes the IPv6 code to still not be covered through CI testing infrastructure - open to thoughts on how to address if this isn't workable.

kenhys commented:
@mcipperly could you rebase tihis pull request?

kenhys commented on fluent/sender.py:
```diff
@@ -201,7 +208,12 @@ def _reconnect(self):
+                    else:
+                        sock = socket.socket(socket.AF_INET6,
+                                             socket.SOCK_STREAM)
                     sock.settimeout(self.timeout)
```
Does it need to call  `_is_ipv4_host()` in every `_make_packet`?
It may be able to reuse checked status.

mcipperly commented:
Updated to address the above re: not wanting to determine v4/v6 capability each time. Now:
* When we connect, save address family
* If we fail to connect in the future, reset address family and try to figure it out again - this could address issues in long-running applications where the target fluent system loses its IPv4 or IPv6 address during runtime

Also added the `prefer_ipv6` option which can be set to `True` to prefer IPv6 and fall back to IPv4
